src/main/java/com/stuypulse/robot/subsystems/Edwin.java

```diff
@@ -0,0 +1,216 @@
+            new CANSparkMax(RIGHT_TOP, MotorType.kBrushless), 
+            new CANSparkMax(RIGHT_BOTTOM, MotorType.kBrushless)
+        };
+
+        LEFT_TOP_MOTOR.configure(left[0]);
+        LEFT_BOTTOM_MOTOR.configure(left[1]);
+
+        RIGHT_TOP_MOTOR.configure(right[0]);
+        RIGHT_BOTTOM_MOTOR.configure(right[1]);
+
+        drivetrain = new DifferentialDrive(
+            new MotorControllerGroup(left), 
+            new MotorControllerGroup(right)
+        );
+
+        leftEncoder = left[0].getEncoder();
+        rightEncoder = right[0].getEncoder();
+
+        leftEncoder.setVelocityConversionFactor(Encoders.GEAR_DISTANCE_PER_ROTATION / 60);
+        rightEncoder.setVelocityConversionFactor(Encoders.GEAR_DISTANCE_PER_ROTATION / 60);
+        
+        leftController = new Feedforward.Drivetrain(Motion.Feedforward.kS, Motion.Feedforward.kV, Motion.Feedforward.kA).velocity()
+            .add(new PIDController(Motion.PID.kP, Motion.PID.kI, Motion.PID.kD));
+        
+        rightController = new Feedforward.Drivetrain(Motion.Feedforward.kS, Motion.Feedforward.kV, Motion.Feedforward.kA).velocity()
+            .add(new PIDController(Motion.PID.kP, Motion.PID.kI, Motion.PID.kD));
+
+        leftTargetSpeed = new SmartNumber("Edwin/Left Target Speed", 0);
+        rightTargetSpeed = new SmartNumber("Edwin/Right TargetSpeed", 0);
+
+        odometry = new DifferentialDriveOdometry(getRotation2d());
+        kinematics = new DifferentialDriveKinematics(TRACK_WIDTH);
+        navx = new AHRS(SPI.Port.kMXP);
+
+        setPose(new Pose2d());
+
+        field = new Field2d();
+    }
+
+    /*********************
+     * ENCODER FUNCTIONS *
+     *********************/
+
+    // Distance
+    private double getLeftDistance() {
+        return leftEncoder.getPosition();
+    }
+
+    private double getRightDistance() {
+        return rightEncoder.getPosition();
+    }
+
+    private double getDistance() {
+        return (getLeftDistance() + getRightDistance()) / 2.0;
+    }
+
+    // Velocity
+    private double getLeftVelocity() {
+        return leftEncoder.getVelocity();
+    }
+
+    private double getRightVelocity() {
+        return rightEncoder.getVelocity();
+    }
+
+    private double getVelocity() {
+        return (getLeftVelocity() + getRightVelocity()) / 2.0;
+    }
+
+    /***************
+     * ROBOT ANGLE *
+     ***************/
+
+    // Gets current Angle of the Robot as a Rotation2d (contiuous / not +-180)
+    public Rotation2d getRotation2d() {
+        return navx.getRotation2d();
+    }
+
+    /**********************
+     * ODOMETRY FUNCTIONS *
+     **********************/
+
+    private void updateOdometry() {
+        odometry.update(getRotation2d(), getLeftDistance(), getRightDistance());
+    }
+
+    public Pose2d getPose() {
+        return odometry.getPoseMeters();
+    }
+
+    @Override
+    public void setPose(Pose2d pose) {
+        odometry.resetPosition(pose, new Rotation2d());
```

Comment:
.resetPosition(pose, getRotation2d()) or this won't work.

Also you MUST reset the position of the encoders in this method.

please **read the documentation of resetPosition**